lib/redcord/base.rb

```diff
@@ -52,8 +52,9 @@ def self.included(klass)
       # coerced to the specified attribute types. Like ActiveRecord,
       # Redcord manages the created_at and updated_at fields behind the
       # scene.
-      prop :created_at, T.nilable(Time)
-      prop :updated_at, T.nilable(Time)
+      attribute :id, T.nilable(Integer), index: true
```

Comment:
Why does id need to be a secondary indexed attribute if it's already the primary key? We could route `Model.where(id)` or `Model.find_by(id)` to `Model.find(id)`. Otherwise, I think this stores unnecessary entries in Redis. E.g. A RedisRecord with an id of 1 would create records:

- `RedisRecord:Model:id:1 => {record data}`  and
- `RedisRecord:Model:attr:id:1 => {1}`

Reply:
> Why does id need to be a secondary indexed attribute if it's already the primary key?

Great question!

Currently we don't have a single set which stores all the ids. It'd be nice to put the ids in a sorted set, so we can perform range queries on those ids.

For example, this would be an useful API:
```ruby
Model.find_in_batches(batch_size: 100) { ... }
```
This is also doable with a SCAN, but that's much less efficient if we have a lot of entries on Redis (O(n) vs O(log(n))

Reply:
Ah thanks, that makes sense! Forgot that integer will be a range index :)